packages/go_router/lib/src/route.dart

```diff
@@ -5,32 +5,141 @@
 import 'package:collection/collection.dart';
 import 'package:flutter/widgets.dart';
 
+import 'configuration.dart';
+import 'pages/custom_transition_page.dart';
 import 'path_utils.dart';
-import 'state.dart';
 import 'typedefs.dart';
 
-/// A declarative mapping between a route path and a page builder.
-class GoRoute {
-  /// Default constructor used to create mapping between a route path and a page
-  /// builder.
+/// The base class for [GoRoute] and [ShellRoute].
+@immutable
+abstract class RouteBase {
+  const RouteBase._({
+    this.routes = const <RouteBase>[],
+  });
+
+  /// The list of child routes associated with this route.
+  ///
+  /// Routes are defined in a tree such that parent routes must match the
```

Comment:
These doc should be in the class doc string

Reply:
55677e9621c4b1b090f8c0a80b97482b598911b0